routers/api/v1/activitypub/person.go

```diff
@@ -90,17 +88,210 @@ func PersonInbox(ctx *context.APIContext) {
 	// ---
 	// summary: Send to the inbox
 	// produces:
-	// - application/json
+	// - application/activity+json
 	// parameters:
 	// - name: username
 	//   in: path
 	//   description: username of the user
 	//   type: string
 	//   required: true
 	// responses:
-	// responses:
 	//   "204":
 	//     "$ref": "#/responses/empty"
 
+	body, err := io.ReadAll(io.LimitReader(ctx.Req.Body, setting.Federation.MaxSize))
+	if err != nil {
+		ctx.ServerError("Error reading request body", err)
+	}
+
+	var activity ap.Activity
+	activity.UnmarshalJSON(body)
+	switch activity.Type {
+	case ap.FollowType:
+		activitypub.Follow(ctx, activity)
+	case ap.UndoType:
+		activitypub.Unfollow(ctx, activity)
+	default:
+		log.Debug("ActivityStreams type not supported", activity)
+		ctx.PlainText(http.StatusNotImplemented, "ActivityStreams type not supported")
+		return
+	}
+
 	ctx.Status(http.StatusNoContent)
 }
+
+// PersonOutbox function returns the user's Outbox OrderedCollection
+func PersonOutbox(ctx *context.APIContext) {
+	// swagger:operation GET /activitypub/user/{username}/outbox activitypub activitypubPersonOutbox
+	// ---
+	// summary: Returns the Outbox OrderedCollection
+	// produces:
+	// - application/activity+json
+	// parameters:
+	// - name: username
+	//   in: path
+	//   description: username of the user
+	//   type: string
+	//   required: true
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/ActivityPub"
+
+	link := setting.AppURL + "api/v1/activitypub/user/" + ctx.ContextUser.Name
+
+	outbox := ap.OrderedCollectionNew(ap.IRI(link + "/outbox"))
+
+	feed, err := models.GetFeeds(ctx, models.GetFeedsOptions{
+		RequestedUser:   ctx.ContextUser,
+		Actor:           ctx.ContextUser,
+		IncludePrivate:  false,
+		IncludeDeleted:  false,
+		ListOptions:     db.ListOptions{Page: 1, PageSize: 1000000},
```

Comment:
Uh for performance reasons we likely want to not do this(Increased pagesize is fine), does AP supports list/pages options like this?

Reply:
AP does support pagination, but the problem is we only want the repo creation actions, but we don't know how many pages of the DB we'll have to query for. Is there a way we can make `models.GetFeeds` only return repo creation actions?

Reply:
https://github.com/go-gitea/gitea/pull/20391/commits/2373b4177ada41cd772b647b0f239887370dee6d is the right direction but we need to implement star action types in another PR so we can remove the stars code.